src/v/cluster/topic_table.cc

```diff
@@ -878,6 +787,105 @@ std::vector<model::ntp> topic_table::all_updates_in_progress() const {
     return ret;
 }
 
+void topic_table::change_partition_replicas(
+  model::ntp ntp,
+  const std::vector<model::broker_shard>& new_assignment,
+  topic_metadata_item& metadata,
+  model::offset o) {
+    auto current_assignment_it = metadata.get_assignments().find(
+      ntp.tp.partition);
+
+    if (are_replica_sets_equal(
+          current_assignment_it->replicas, new_assignment)) {
```

Comment:
Caller should do all this check before

Reply:
it is ok to have the caller check. it's like a pre-condition on the method. but it is good practice to document your pre-conditions. it doesn't need to be formal: it can be an assertion plus a comment stating the caller guarantees it.